lib/puppet/util/windows/service.rb

```diff
@@ -558,16 +561,17 @@ def open_scm(scm_access, &block)
       #
       # @param [String] service_name the name of the service to transition
       # @param [[Integer]] valid_initial_states an array of valid states that the service can transition from
-      # @param [:Integer] final_state the state that the service will transition to
-      def transition_service_state(service_name, valid_initial_states, final_state, &block)
+      # @param [Integer] final_state the state that the service will transition to
+      # @param [Integer] timeout the minumum number of seconds to wait before timing out
+      def transition_service_state(service_name, valid_initial_states, final_state, timeout, &block)
```

Comment:
Nit: It would be nice to have the `timeout` be an option here and in the `wait_` helpers as well to keep things consistent.

Reply:
I can change it, but it doesn't really make sense to have the private functions use an optional timeout. Since the functions are private, any use of their functionality would need to go through a public function. Since any public function that uses these helpers will already have an optional timeout it's unnecessary to have another optional timeout in the private versions.

To me it's not inconsistent that the public functions have optional params and the private ones don't, they are just different.